src/bootstrap/bin/main.rs

```diff
@@ -16,17 +20,24 @@ fn main() {
     let mut build_lock;
     let _build_lock_guard;
     if cfg!(any(unix, windows)) {
+        // Get existing contents of lock file
         let path = config.out.join("lock");
+        let mut pid = Vec::new();
+        if path.exists() {
+            let mut _lock_file = t!(std::fs::File::open(&path));
```

Comment:
Rather than testing if the path exists before opening the file, why not try to open the file and then handle any `ErrorKind::NotFound`? This handles a race condition where the file is deleted after `path.exists()` returns try but before `File::open` is called.

Reply:
Okay! Will make the necessary changes!(: Thank you!

Also, out of curiosity, in what situations would the race condition occur? i.e. when would a file be deleted after path.exists() but before the file is opened?

Reply:
Well the filesystem is completely independent of the application. The file could be deleted by another process on the system at any time at all before we actually open it. And if that delete just so happens to occur right after the `exists()` call then we'll get an error. Admittedly in this case it's probably not the biggest deal in the world but in other cases it can be a security issue.

The phrase to search for here is "time-of-check to time-of-use" (or TOCTOU for short).